packages/@ourworldindata/grapher/src/stackedCharts/AbstractStackedChart.tsx

```diff
@@ -373,20 +379,37 @@ export class AbstractStackedChart
             (shortUnit) => shortUnit === "%"
         )
 
-        // Normally StackedArea/StackedBar charts are always single-entity or single-column, but if we are ever in a mode where we
-        // have multiple entities selected (e.g. through GlobalEntitySelector) and multiple columns, it only makes sense when faceted.
-        if (!this.isEntitySeries && !areMultipleEntitiesSelected)
-            strategies.push(FacetStrategy.none)
-        else if (this.isEntitySeries && !hasMultipleYColumns)
+        // Normally StackedArea/StackedBar charts are always single-entity or single-column,
+        // but if we are ever in a mode where we have multiple entities selected (e.g. through
+        // GlobalEntitySelector) and multiple columns, it only makes sense when faceted.
+        if (
+            // No facet strategy makes sense if columns are stacked and a single entity is selected
+            (!this.isEntitySeries && !areMultipleEntitiesSelected) ||
+            // No facet strategy makes sense if entities are stacked and we have a single column
+            (this.isEntitySeries &&
+                !hasMultipleYColumns &&
+                // The stacking must be sensible
+                checkIsStackingEntitiesSensible(selectedEntityNames))
+        )
             strategies.push(FacetStrategy.none)
```

Comment:
One thing I'm wondering here is if we can ever run into the scenario where we end up returning an empty array.

And we rely on this fact here: https://github.com/owid/owid-grapher/blob/aee8d4a17d5fe520d00fbfa42a52408811232a70/packages/%40ourworldindata/grapher/src/core/Grapher.tsx#L2944

In the past, this was either highly unlikely or completely impossible.

Now, in theory, it would be possible:
E.g. when selecting "World" and "Europe", and the chart has multiple units.

Probably, we don't have any charts where that's possible, but it would at least be good to make us aware of such issues. Could you throw an error in that case, and send that error to Sentry?

Reply:
Yeah, that's a good point! I'll add error handling.